src/work-allocation/components/task-assignment/task-assignment.component.ts

```diff
@@ -83,17 +86,27 @@ export class TaskAssignmentComponent implements OnInit {
   private pExcludeCaseworkers: Caseworker[];
   private readonly caseworkerDisplayName: CaseworkerDisplayName = new CaseworkerDisplayName();
 
+  public userId: string;
```

Comment:
Why are these `public` properties? Is it purely for unit testing as I don't see them being used anywhere other than the spec file?

If they're _only_ public for unit test purposes, I think some refactoring is needed. The same applies to some of the public methods below. Methods and properties shouldn't be public purely for the benefit of unit testing as it means you're exposing internal workings - and things that are being _used_ as private properties - when you didn't necessarily intend to.

You could, for example, add a getter method for a `private pUserId: string;` property if you need to access it in the unit tests. That, at the least, won't make the private property editable externally. Also, you have a `setUserId()` method, which making this public essentially circumvents. That method is _also_ `public`, which looks less concerning but still appears to be purely for unit testing purposes.

Reply:
Yes, I agree with that. I'll rework the public properties into private ones (similar to the other properties within the file) and refactor the tests.